lib/component.js

```diff
@@ -327,6 +361,13 @@ class Component extends WebComponent {
       return;
     }
 
+    for (let i = 0; i < this._disconnectedQueue.length; i++) {
+      const disconnectedCallbackFn = this._disconnectedQueue[i];
+      disconnectedCallbackFn.call(this);
```

Comment:
We probably need to try/catch this and L352 above. Since these are arbitrary callbacks we'd like to limit the blast radius if they throw exceptions, and continue to clean up the component gracefully.